### PR TITLE
Case specific results summary in dashboard filters

### LINKED ISSUE #1438: Improve user visible strings and fix localization of `_filters.html`
Currently the template looks like this:

```django
<span>{% trans 'Total records found' context 'Dashboard filter' %}: {{ count }}</span>
```

The message is generic and uses the technical term "records" to avoid having to know what type of items is filtered. I think we can do better and provide use case-specific messages that don't look awkward and are easier to translate.

Another problem is that the translation does not include the counter so it's impossible to properly translate it for RTL languages.

Yet another problem is that the translation does not depend on the counter so it's impossible to provide correct grammatical alternatives for various values of the counter.

A good solution would look more like this:

```django
<span>{% blocktrans count counter=count context 'Number of matching product records in the dashboard list of product' %}Found {{ count }} matching products{% endblocktrans %}</span>
```

### PR DESCRIPTION
**Pull request by mad-anne**
We had generic filters summary sentence. Now it is specific case and based on queryset we are filtering, e. g. ```Found  1 matching product```.

Closes #1438

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [x] JavaScript code quality checks pass: `eslint`.

### CODE REVIEW

File: saleor/dashboard/category/views.py [hunk 1]
```diff
@@ -22,7 +22,14 @@ def category_list(request):
     categories = get_paginator_items(
         category_filter.qs, settings.DASHBOARD_PAGINATE_BY,
         request.GET.get('page'))
-    ctx = {'categories': categories, 'filter': category_filter}
+    summary_msg = npgettext_lazy(
+        'Filter set results summary',
+        'Found %(counter)d matching category',
+        'Found %(counter)d matching categories',
+        'counter') % {'counter': category_filter.qs.count()}
```
Comment (patrys) [inline saleor/dashboard/category/views.py]: What is the word `counter` doing here as the `number` parameter?

Comment (patrys) [reply saleor/dashboard/category/views.py]: Which brings me to another question, why is it a lazy call? 😄

File: saleor/dashboard/category/views.py [hunk 2]
```diff
@@ -22,7 +22,14 @@ def category_list(request):
     categories = get_paginator_items(
         category_filter.qs, settings.DASHBOARD_PAGINATE_BY,
         request.GET.get('page'))
-    ctx = {'categories': categories, 'filter': category_filter}
+    summary_msg = npgettext_lazy(
+        'Filter set results summary',
```
Comment (patrys) [inline saleor/dashboard/category/views.py]: Could we come up with a more descriptive context message here?

Comment (mad-anne) [reply saleor/dashboard/category/views.py]: Yes, I changed it and message relates now to what objects are filtered.

File: saleor/dashboard/category/views.py (hunk 1 above)
Comment (mad-anne) [reply saleor/dashboard/category/views.py]: Changed it to simple ```npgettext```.

**Pull request merged**